libs/gamify/badges/src/lib/badges.controller.ts

```diff
@@ -0,0 +1,65 @@
+import { ApplicationsService } from '@gamify/application';
+import { BadRequestException, Body, Controller, Delete, Get, NotFoundException, Param, ParseIntPipe, Post, Put } from '@nestjs/common';
+import { BadgesService } from './badges.service';
+import { CreateBadgeInput } from './dto/create-badge.input';
+import { UpdateBadgeInput } from './dto/update-badge.input';
+import { BadgeModel } from './models';
+
+@Controller('badges')
+export class BadgesController {
+  constructor(private badgesService: BadgesService, private applicationsService: ApplicationsService) {}
+
+  @Post()
+  async create(@Body() createBadgeInput: CreateBadgeInput): Promise<BadgeModel> {
+    if (await this.applicationsService.findOne(createBadgeInput.applicationId) === null) {
+      throw new BadRequestException('Application not found');
+    }
+
+    // TODO check authorization of user.
+    return this.badgesService.create(createBadgeInput);
+  }
+
+  @Get()
+  findAll(): Promise<BadgeModel[]> {
+    return this.badgesService.findMany();
+  }
+
+  @Get(':id')
+  findOne(@Param('id', ParseIntPipe) id: number): Promise<BadgeModel> {
+    return this.findBadgeOrFail(id);
+  }
+
+  @Put(':id')
+  async update(@Param('id', ParseIntPipe) id: number, @Body() updateBadgeInput: UpdateBadgeInput): Promise<BadgeModel> {
+    // TODO check authorization of user.
+
+    await this.findBadgeOrFail(id);
+
+    return this.badgesService.update(id, updateBadgeInput);
+  }
+
+  @Delete(':id')
+  async remove(@Param('id', ParseIntPipe) id: number): Promise<BadgeModel> {
```

Comment:
Does and should `DELETE` return the deleted object `BadgeModel`?
(Same for the underlying `this.badgesService.remove(id)`)

Reply:
Yes it returns the deleted object `BadgeModel`, I'm not exactly sure if this is the best behaviour but going to leave it like this for now